### PR TITLE
Feature/72 question3.4 

### LINKED ISSUE #72: Question 3.4 inline with 3.1, 3.2, 3.3
Question 3.4 will be merged in with questions 3.1, 3.2, and 3.3. There will be no more question 3.4

Question 3.1, 3.2, and 3.3 are all different types of aims. When the user selects one of these aims, a new question will appear underneath the selected checkbox. List the responses from Q2.1 (leave empty if there are no responses in Q2.1) with the question  text from 3.4: "identify which are primary, or co-benefits"

For each of the selected stakeholders selected in Q2.1, show a dropdown with the options:
- Primary aims
- Co-benefits
- Neither

TODO: Notify the user if they've already filled out question 3.1, 3.2, or 3.3 then add a new selection to Q2.1.

Image

### PR DESCRIPTION
**Pull request by mmnoo**
# Description

- incorporate question 3.4 into 3.1, 3.2, 3.3
- modify causes of decline form to use MUI FormControlLabel and to avoid common component in `forms.js` to be confused with other used in nested questions
- make sure when user unselects parent checkbox, nested values clear
- loads all values from api
- validations still works
- no format change for CheckboxGroup (like previous WIP contained)

Fixes #72

### CODE REVIEW

File: mrtt-ui/src/library/forceIsOtherCheckedPropertyToBeBooleanBecauseYupIsntGreat.js [hunk 1]
```diff
@@ -0,0 +1,3 @@
+export const forceIsOtherCheckedPropertyToBeBooleanBecauseYupIsntGreat = (object) => {
```
Comment (mmnoo) [inline mrtt-ui/src/library/forceIsOtherCheckedPropertyToBeBooleanBecauseYupIsntGreat.js:1]: this is from a previous abandoned approach. Just not ready to throw away yet :)

File: mrtt-ui/src/validation/multiSelectWithOther.js [hunk 2]
```diff
@@ -6,23 +6,32 @@ export const otherIsCheckedButInputIsEmptyValidation = (schema) =>
...
     .array()
-    .of(yup.string())
+    .default([])
     .when('isOtherChecked', {
-      is: false || undefined,
-      then: (schema) =>
-        schema.min(1, language.multiselectWithOtherFormQuestion.validation.selectAtleastOneItem),
+      is: (value) => {
+        console.log(value)
+        return value === false // doesnt play well with undefined, or !value. We need to explicitly set isOtherChecked to false if undefined.
```
Comment (mmnoo) [inline mrtt-ui/src/validation/multiSelectWithOther.js:24]: this is why we need the .default(false) and also weirdly the .default([]). Wasnt in the docs that it wont interpret an undefined, even if I set it with !!value.  Feels like a bug.

File: mrtt-ui/src/components/RestorationAimsForm/RestorationAimsForm.js [hunk 3]
```diff
@@ -13,27 +13,30 @@ import { multiselectWithOtherValidation } from '../../validation/multiSelectWith
...
     otherAims: multiselectWithOtherValidation
   })
   const reactHookFormInstance = useForm({
     defaultValues: {
-      ecologicalAims: { selectedValues: [], otherValue: undefined },
-      socioEconomicAims: { selectedValues: [], otherValue: undefined },
-      otherAims: { selectedValues: [], otherValue: undefined }
+      // ecologicalAims: { selectedValues: [], otherValue: undefined }
+      // socioEconomicAims: { selectedValues: [], otherValue: undefined },
+      otherAims: { selectedValues: [], otherValue: undefined, isOtherChecked: false }
```
Comment (mmnoo) [inline mrtt-ui/src/components/RestorationAimsForm/RestorationAimsForm.js:39]: this didnt help, because when we load stuff from the api, isOtherChecked gets set back to undefined.

File: mrtt-ui/src/components/CheckboxGroup/CheckboxGroup.stories.js [hunk 4]
```diff
@@ -34,8 +34,25 @@ export const OtherOptionPrePopulated = () => (
...
     shouldAddOtherOptionWithClarification={true}
   />
 )
+
+export const WithNestedMarkupForSelected = () => (
+  <CheckboxGroup
+    options={options}
+    value={{
+      selectedValues: [{ choice: 'Option 1' }, { choice: 'Option 2' }],
+      otherValue: { choice: 'Some other value' }
```
Comment (mmnoo) [inline mrtt-ui/src/components/CheckboxGroup/CheckboxGroup.stories.js:51]: this format is working, and will let us add properties for nested answers.

File: mrtt-ui/src/components/CheckboxGroup/CheckboxGroup.stories.js [hunk 5]
```diff
@@ -34,8 +34,25 @@ export const OtherOptionPrePopulated = () => (
...
+    }}
+    onChange={action('onChange')}
+    id='cuddlyKittens'
+    shouldAddOtherOptionWithClarification={true}
+    selectedMarkup={<>Youve selected ME</>}
```
Comment (mmnoo) [inline mrtt-ui/src/components/CheckboxGroup/CheckboxGroup.stories.js:56]: selectedMarkup allows the parent to inject and controll nested questions. the component will append an unique `optionId` for each iteration of the selectedMarkup (it gets applied to each option if selected)

File: mrtt-ui/src/components/RestorationAimsForm/RestorationAimsCheckboxGroupWithLabel.js [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+              id={fieldName}
+              defaultValue={{ selectedValues: [], otherValue: undefined, isOtherChecked: false }}
+              SelectedMarkup={({ optionId }) => (
+                <StakeholderBenefitsInputs
+                  stakeholders={stakeholders}
+                  reactHookFormInstance={reactHookFormInstance}
+                  parentQuestionFieldName={fieldName}
+                  optionId={optionId}
+                />
+              )}
```
Comment (mmnoo) [inline mrtt-ui/src/components/RestorationAimsForm/RestorationAimsCheckboxGroupWithLabel.js:52]: proptypes doesnt like this partially applied component, so I allowed an `any`

File: mrtt-ui/src/components/RestorationAimsForm/StakeholderBenefitsInputs.js [hunk 7]
```diff
@@ -0,0 +1,86 @@
...
+  PropTypes.shape({ stakeholderType: PropTypes.string, stakeholderName: PropTypes.string })
+)
+
+const StakeholderBenefitsInputs = ({
+  optionId,
+  parentQuestionFieldName,
+  reactHookFormInstance,
+  stakeholders
+}) => {
+  const questionLabelId = `${optionId}-question-label`
```
Comment (mmnoo) [inline mrtt-ui/src/components/RestorationAimsForm/StakeholderBenefitsInputs.js:21]: we need the unique optionId generated based on options (from checkboxgroup) for accessibility. Each nested question needs a uniqueid.

File: mrtt-ui/src/components/RestorationAimsForm/StakeholderBenefitsInputs.js [hunk 8]
```diff
@@ -0,0 +1,86 @@
...
+  ))
+
+  return (
+    <NestedQuestion>
+      <FormLabel id={questionLabelId}>
+        {questions.nestedStakeholderBenefitsRanking.question}
+      </FormLabel>
+      <div aria-labelledby={questionLabelId}>
+        {stakeholdersToUse.map(({ stakeholderType, stakeholderName }) => {
+          const selectInputId = makeValidClassName(`${optionId}${stakeholderType}-input`)
```
Comment (mmnoo) [inline mrtt-ui/src/components/RestorationAimsForm/StakeholderBenefitsInputs.js:52]: this id ends up looking like aimNameStakeholderType

Comment (mmnoo) [reply mrtt-ui/src/components/RestorationAimsForm/StakeholderBenefitsInputs.js:52]: and ends up being a property on the parent data. aimNameStakeholderType: values

Comment (mmnoo) [reply mrtt-ui/src/components/RestorationAimsForm/StakeholderBenefitsInputs.js:52]: or probably nested within another property I willl call `stakeholderBenefits`. Not sure on if I implemented that or not yet.

File: mrtt-ui/src/components/RestorationAimsForm/StakeholderBenefitsInputs.js [hunk 9]
```diff
@@ -0,0 +1,86 @@
...
+    getValues: getFormValues,
+    control: formControl
+  } = reactHookFormInstance
+  const benefitsFieldName = `${parentQuestionFieldName}StakeholderBenefits`
+  const stakeholderBenefits = useWatch({ control: formControl, name: benefitsFieldName })
+
+  const handleOnChange = ({ event, stakeholderPropertyName }) => {
+    const previousBenefits = getFormValues(benefitsFieldName)
+
+    setFormValue(benefitsFieldName, {
```
Comment (mmnoo) [inline mrtt-ui/src/components/RestorationAimsForm/StakeholderBenefitsInputs.js:34]: benefitsFieldName is the parent fieldName. Se we are appending the nested answers to the parent here

File: mrtt-ui/src/components/RestorationAimsForm/RestorationAimsCheckboxGroupWithLabel.js [hunk 10]
```diff
@@ -0,0 +1,71 @@
...
+  question,
+  reactHookFormInstance,
+  stakeholders
+}) => {
+  const { control: formControl, setValue: setFormValue, getValues } = reactHookFormInstance
+  const id = `${fieldName}-label`
+  const optionsValueLabels = getOptionsValuesAndLabels(options)
+
+  const handleAimChange = ({ selectedValues, otherValue, isOtherChecked }) => {
+    setFormValue(fieldName, { selectedValues, otherValue, isOtherChecked })
```
Comment (mmnoo) [inline mrtt-ui/src/components/RestorationAimsForm/RestorationAimsCheckboxGroupWithLabel.js:23]: Not sure if this code does what I intend yet. Got distracted by validation.

File: mrtt-ui/src/components/CausesOfDeclineForm.js [hunk 11]
```diff
@@ -288,21 +287,21 @@ function CausesOfDeclineForm() {
...
-                                          mainCauseLabel: mainCause.label,
-                                          subCauseLabel: subCause.secondaryLabel,
-                                          secondaryChildOption
-                                        })
-                                      }></Checkbox>
-                                    <Typography variant='subtitle2'>
-                                      {secondaryChildOption}{' '}
-                                    </Typography>
-                                  </NestedFormSectionDiv>
+                                  <FormControlLabel
```
Comment (mmnoo) [inline mrtt-ui/src/components/CausesOfDeclineForm.js:290]: I noticed this when I noticed that your reusable NestedFormSectionDiv is easily confused with my new NestedQuestion. Tried to figure out which one to use, and then realized yours could be solved with FormControlLabel. So took the liberty to change it.

File: mrtt-ui/src/components/CheckboxGroup/CheckboxGroup.js [hunk 12]
```diff
@@ -1,36 +1,41 @@
...
+import { Checkbox, FormControlLabel, FormGroup, Stack, TextField } from '@mui/material'
 import PropTypes from 'prop-types'
 import { forwardRef, useEffect, useState } from 'react'
 import language from '../../language'
+import { makeValidClassName } from '../../library/strings/makeValidClassName'
 
-const getValueToReturn = ({ nonOtherSelectedValues, otherInputValue, isOtherChecked }) => {
-  const valueToReturn = { selectedValues: nonOtherSelectedValues ?? [] }
+const getValueToReturn = ({ nonOtherSelectedValues, otherInputValue, isOtherChecked, event }) => {
+  const valueToReturn = { selectedValues: nonOtherSelectedValues ?? [], event }
```
Comment (mmnoo) [inline mrtt-ui/src/components/CheckboxGroup/CheckboxGroup.js:8]: adding original event to object being returned so we can grab its value for the resetting of nested answers when an item is unchecked (because the data format required the option value). see `handleAimChangeAndResetStakeholdersIfAppropriate`

File: mrtt-ui/src/components/CheckboxGroup/CheckboxGroup.stories.js [hunk 13]
```diff
@@ -39,3 +39,14 @@ export const OtherOptionPrePopulated = () => (
...
 )
+
+export const WithNestedMarkupForSelected = () => (
+  <CheckboxGroup
+    options={options}
+    value={{ selectedValues: ['Option 1', 'Option 2'], otherValue: 'Some other value' }}
+    onChange={action('onChange')}
+    id='cuddlyKittens'
+    shouldAddOtherOptionWithClarification={true}
+    SelectedMarkup={() => <>Youve selected ME</>}
```
Comment (mmnoo) [inline mrtt-ui/src/components/CheckboxGroup/CheckboxGroup.stories.js:50]: SelectedMarkup is basically the big change to CheckboxGroup. Whatever you put here, will show for any checkbox checked (always the same markup - wont work for nested stuff that is unique for each checkbox)

File: mrtt-ui/src/components/RestorationAimsForm/RestorationAimsCheckboxGroupWithLabel.js [hunk 14]
```diff
@@ -0,0 +1,92 @@
...
+}) => {
+  const {
+    control: formControl,
+    getValues: getFormValues,
+    setValue: setFormValue
+  } = reactHookFormInstance
+  const optionsValueLabels = getOptionsValuesAndLabels(options)
+  const id = `${fieldName}-label`
+
+  const handleAimChangeAndResetStakeholdersIfAppropriate = ({
```
Comment (mmnoo) [inline mrtt-ui/src/components/RestorationAimsForm/RestorationAimsCheckboxGroupWithLabel.js:26]: this is where the parent uses event from CheckboxGroup

File: mrtt-ui/src/components/RestorationAimsForm/RestorationAimsCheckboxGroupWithLabel.js [hunk 15]
```diff
@@ -0,0 +1,92 @@
...
+    setFormValue(fieldName, {
+      selectedValues,
+      otherValue,
+      isOtherChecked,
+      aimStakeholderBenefits: isAimChecked
+        ? aimStakeholderBenefitsInForm
+        : aimStakeholderBenefitsReset
+    })
+  }
+
```
Comment (mmnoo) [inline mrtt-ui/src/components/RestorationAimsForm/RestorationAimsCheckboxGroupWithLabel.js:48]: hard to tell the data format from this, but if you look in postman, its be best I could come up with the data analyst in mind given the time constraints

File: mrtt-ui/src/components/RestorationAimsForm/RestorationAimsCheckboxGroupWithLabel.js [hunk 16]
```diff
@@ -0,0 +1,92 @@
...
+        <FormLabel id={id}>{question}</FormLabel>
+      </FormQuestionDiv>
+      <Controller
+        name={fieldName}
+        control={formControl}
+        render={({ field }) => {
+          return (
+            <CheckboxGroup
+              {...field}
+              onChange={handleAimChangeAndResetStakeholdersIfAppropriate}
```
Comment (mmnoo) [inline mrtt-ui/src/components/RestorationAimsForm/RestorationAimsCheckboxGroupWithLabel.js:61]: override onchange and handle it ourselves

File: mrtt-ui/src/components/RestorationAimsForm/StakeholderBenefitsInputs.js [hunk 17]
```diff
@@ -0,0 +1,95 @@
+import { useWatch } from 'react-hook-form'
```
Comment (mmnoo) [inline mrtt-ui/src/components/RestorationAimsForm/StakeholderBenefitsInputs.js:1]: This is a component I made to be given to CheckboxGroup for the nested questions

File: mrtt-ui/src/components/SiteBackgroundForm.js [hunk 18]
```diff
@@ -153,7 +154,7 @@ const ProjectDetailsForm = () => {
             control={control}
             defaultValue=''
             render={({ field }) => (
-              <TextField {...field} select value={field.value} label='select'>
+              <TextField {...field} select value={field.value} label={language.form.selectLabel}>
```
Comment (mmnoo) [inline mrtt-ui/src/components/SiteBackgroundForm.js:157]: just language reuse here

File: mrtt-ui/src/library/formatApiAnswersForForm.js [hunk 19]
```diff
@@ -7,7 +7,7 @@ const formatApiAnswersForForm = ({ apiAnswers, questionMapping }) => {
     const [frontEndAnswerId, apiAnswerId] = question
     const apiAnswer = apiAnswers.find((apiAnswer) => apiAnswer.question_id === apiAnswerId)
 
-    return [frontEndAnswerId, apiAnswer.answer_value]
+    return [frontEndAnswerId, apiAnswer?.answer_value]
```
Comment (mmnoo) [inline mrtt-ui/src/library/formatApiAnswersForForm.js:10]: bug fix for when no data from api

File: mrtt-ui/src/validation/multiSelectWithOther.js [hunk 20]
```diff
@@ -16,13 +16,13 @@ export const otherIsCheckedButInputIsEmptyValidation = (schema) =>
 export const multiselectWithOtherValidation = yup.object({
```
Comment (mmnoo) [inline mrtt-ui/src/validation/multiSelectWithOther.js:16]: validation bug fix for when item value is undefined. Set default in yup schema.

File: mrtt-ui/src/components/CheckboxGroup/CheckboxGroup.js [hunk 21]
```diff
@@ -127,15 +146,19 @@ CheckboxGroup.propTypes = {
...
-    otherValue: PropTypes.string
+    otherValue: PropTypes.string,
+    isOtherChecked: PropTypes.bool
   })
 }
 
 CheckboxGroup.defaultProps = {
   onBlur: () => {},
+  value: [],
+  SelectedMarkup: undefined,
```
Comment (julia-rom) [inline mrtt-ui/src/components/CheckboxGroup/CheckboxGroup.js:161]: what is` SelectedMarkup`?

File: mrtt-ui/src/components/CheckboxGroup/CheckboxGroup.stories.js [hunk 22]
```diff
@@ -39,3 +39,14 @@ export const OtherOptionPrePopulated = () => (
     shouldAddOtherOptionWithClarification={true}
   />
 )
+
+export const WithNestedMarkupForSelected = () => (
+  <CheckboxGroup
+    options={options}
+    value={{ selectedValues: ['Option 1', 'Option 2'], otherValue: 'Some other value' }}
+    onChange={action('onChange')}
+    id='cuddlyKittens'
```
Comment (julia-rom) [inline mrtt-ui/src/components/CheckboxGroup/CheckboxGroup.stories.js:48]: cute id name but why is this here 🐱?

File: mrtt-ui/src/components/CheckboxGroup/CheckboxGroup.stories.js (hunk 13 above)
Comment (julia-rom) [reply mrtt-ui/src/components/CheckboxGroup/CheckboxGroup.stories.js:50]: hmm what would be a big change? would this be like my section 4 deep nesting structure? what would be an example of how to use this?

**Review by julia-rom — COMMENTED** [review]
This is looking good! I have some questions before I approve, just want to make sure I understand the checkbox group changes as well as `useInitializeQuestionMappedForm`

File: mrtt-ui/src/components/CausesOfDeclineForm.js [hunk 23]
```diff
@@ -288,21 +287,21 @@ function CausesOfDeclineForm() {
...
+                                      <Checkbox
+                                        value={secondaryChildOption}
+                                        onChange={(event) =>
+                                          handleCausesOfDeclineOnChange({
+                                            event,
+                                            mainCauseLabel: mainCause.label,
+                                            subCauseLabel: subCause.secondaryLabel,
+                                            secondaryChildOption
+                                          })
+                                        }></Checkbox>
```
[the remark this answers is not in the document]
Comment (mmnoo) [reply mrtt-ui/src/components/CausesOfDeclineForm.js:301]: honestly, its been a long time so I cant remember, but I think we get the label styling for free from FormControlLabel

File: mrtt-ui/src/components/CheckboxGroup/CheckboxGroup.js [hunk 24]
```diff
@@ -1,36 +1,39 @@
...
 }
 
 const CheckboxGroup = forwardRef(
-  ({ id, onBlur, onChange, options, shouldAddOtherOptionWithClarification, value }, ref) => {
-    const [nonOtherSelectedValues, setNonOtherSelectedValues] = useState([])
+  (
+    { id, onBlur, onChange, options, shouldAddOtherOptionWithClarification, value, SelectedMarkup },
+    ref
+  ) => {
+    const [isOtherChecked, setIsOtherChecked] = useState(!!value?.isOtherChecked)
```
[the remark this answers is not in the document]
Comment (mmnoo) [reply mrtt-ui/src/components/CheckboxGroup/CheckboxGroup.js:22]: I guess I just like that better than having it as undefined if its not true and hasnt been explicitly set in the value. Pretty sure it works either way, but changing it means I have to test it all over again to be super sure.

File: mrtt-ui/src/components/CheckboxGroup/CheckboxGroup.js [hunk 25]
```diff
@@ -62,54 +66,69 @@ const CheckboxGroup = forwardRef(
         getValueToReturn({
```
[the remark this answers is not in the document]
Comment (mmnoo) [reply mrtt-ui/src/components/CheckboxGroup/CheckboxGroup.js:66]: I cant remember. It was important to return to the consumer for something. Maybe yup validation because an empty string in `otherValue` will come through as true if you test for that?

File: mrtt-ui/src/components/CheckboxGroup/CheckboxGroup.stories.js (hunk 22 above)
Comment (mmnoo) [reply mrtt-ui/src/components/CheckboxGroup/CheckboxGroup.stories.js:48]: we need an id to create a unique optionId for each option and the label for each option vs the group of checkboxes as a whole to make them accessible. Otherwise with no id, if we have two checkbox groups on a single page, we get non-unique option ids and the htmlFors get weird.

File: mrtt-ui/src/components/CheckboxGroup/CheckboxGroup.stories.js (hunk 13 above)
Comment (mmnoo) [reply mrtt-ui/src/components/CheckboxGroup/CheckboxGroup.stories.js:50]: See `mrtt-ui/src/components/RestorationAimsForm/StakeholderBenefitsInputs.js` and `mrtt-ui/src/components/RestorationAimsForm/RestorationAimsCheckboxGroupWithLabel.js`. You can also run storybook and take a look. This story is the most basic example. 

this is basically inversion of control so that we can control the nested children for each option from a composing parent - to keep checkbox group generic and reusable (although lets be honest, its probably reaching the max limit for complexity for a generic)

**Review by mmnoo — COMMENTED** [review]
Good questions

File: mrtt-ui/src/library/useInitializeQuestionMappedForm.js [hunk 26]
```diff
@@ -5,26 +5,36 @@ import { useEffect } from 'react'
...
             setIsLoading(false)
             const initialValuesForForm = formatApiAnswersForForm({
-              apiAnswers: data,
+              apiAnswers: response.data,
               questionMapping
             })
             resetForm(initialValuesForForm)
+            if (successCallback) {
+              successCallback(response)
+            }
```
Comment (mmnoo) [inline mrtt-ui/src/library/useInitializeQuestionMappedForm.js:30]: added the conditional so this doesnt break if there is no successCallback

File: mrtt-ui/src/components/CheckboxGroup/CheckboxGroup.stories.js (hunk 13 above)
Comment (julia-rom) [reply mrtt-ui/src/components/CheckboxGroup/CheckboxGroup.stories.js:50]: ok cool! yes I agree that is probably reaching max complexity

File: mrtt-ui/src/components/CheckboxGroup/CheckboxGroup.js (hunk 24 above)
[the remark this answers is not in the document]
Comment (julia-rom) [reply mrtt-ui/src/components/CheckboxGroup/CheckboxGroup.js:22]: ok cool! thanks for explaining, agree it's better than undefined

File: mrtt-ui/src/components/CheckboxGroup/CheckboxGroup.stories.js (hunk 22 above)
Comment (julia-rom) [reply mrtt-ui/src/components/CheckboxGroup/CheckboxGroup.stories.js:48]: ok, can we rename it to `option` or something specific to it?

**Comment by julia-rom** [comment]
I am good to approve this - just a small tweak about renaming one of the ids @mmnoo

File: mrtt-ui/src/components/CheckboxGroup/CheckboxGroup.js (hunk 25 above)
[the remark this answers is not in the document]
Comment (mmnoo) [reply mrtt-ui/src/components/CheckboxGroup/CheckboxGroup.js:66]: I am realizing that I answered not your question, but some other question. . If we uncheck other, we leave the input in the front end..... BUT it might not get submitted..... I need to check this again to make sure it makes sense. I vaguely remember making a cut corner deliberately for this, but its been so long.

File: mrtt-ui/src/components/CheckboxGroup/CheckboxGroup.stories.js (hunk 22 above)
Comment (mmnoo) [reply mrtt-ui/src/components/CheckboxGroup/CheckboxGroup.stories.js:48]: I like using id. People know it needs to be unique, and understand what it is right away. Since its used to generate other unique ids, it fits.

Comment (julia-rom) [reply mrtt-ui/src/components/CheckboxGroup/CheckboxGroup.stories.js:48]: using id is fine, I just mean can we rename something other than `cuddlyKittens` @mmnoo
Screen Shot 2022-06-17 at 12 56 13 PM

Comment (mmnoo) [reply mrtt-ui/src/components/CheckboxGroup/CheckboxGroup.stories.js:48]: We could, but its a bit too much of a nitpick IMO. This is a story for demo, not code where the meaning of the id needs to be descriptive and void of personality. Its totally ok to use random silly values for mock data that only get used in tests for example (as long as its not offensive, racist, mean spirited, etc.) 

That is to say, I get the rule you are referring to, I just dont think it is appropriately applied here.

Comment (julia-rom) [reply mrtt-ui/src/components/CheckboxGroup/CheckboxGroup.stories.js:48]: I missed that this was in stories earlier! sorry about that - that's fine with me 👍

File: mrtt-ui/src/components/CheckboxGroup/CheckboxGroup.js (hunk 25 above)
[the remark this answers is not in the document]
Comment (mmnoo) [reply mrtt-ui/src/components/CheckboxGroup/CheckboxGroup.js:66]: unchecking other will clear the stakeholder benefits selects, but not the other value _in the ui_. Unchecking other will make sure the other value and the stakeholder benefits sent to the api are undefined/empty. It was a shortcut for timelines.  

In exploring this I found and fixed a bug! will add comment when I push it.

File: mrtt-ui/src/components/CheckboxGroup/CheckboxGroup.js [hunk 27]
```diff
@@ -1,36 +1,59 @@
...
+}) => {
   const valueToReturn = { selectedValues: nonOtherSelectedValues ?? [] }
+  valueToReturn.isOtherChecked = !!isOtherChecked //we're being explicit here for the sake of composing code not having to use empty strings vs undefined to differentiate between checked and unchecked for validation for example.
   if (isOtherChecked) {
     valueToReturn.otherValue = otherInputValue
-    valueToReturn.isOtherChecked = true //we're being explicit here for the sake of composing code not having to use empty strings vs undefined to differentiate between checked and unchecked for validation for example.
+  }
+  if (shouldReturnEvent) {
+    // react hook form doesnt like the event being attached to the value object if it handles onchange, so made a flag for when we are handling onchange manually and want to use the event
+    valueToReturn.event = event
```
Comment (mmnoo) [inline mrtt-ui/src/components/CheckboxGroup/CheckboxGroup.js:21]: fixed bug I found.

Comment (mmnoo) [reply mrtt-ui/src/components/CheckboxGroup/CheckboxGroup.js:21]: Im merging now, but please ping me in slack if you have any comments on this bugfix.
